google/resource_compute_address.go

```diff
@@ -97,17 +149,38 @@ func resourceComputeAddressCreate(d *schema.ResourceData, meta interface{}) erro
 }
 
 func resourceComputeAddressRead(d *schema.ResourceData, meta interface{}) error {
+	computeApiVersion := getComputeApiVersion(d, AddressBaseApiVersion, AddressVersionedFeatures)
 	config := meta.(*Config)
 
 	addressId, err := parseComputeAddressId(d.Id(), config)
 	if err != nil {
 		return err
 	}
 
-	addr, err := config.clientCompute.Addresses.Get(
-		addressId.Project, addressId.Region, addressId.Name).Do()
-	if err != nil {
-		return handleNotFoundError(err, d, fmt.Sprintf("Address %q", d.Get("name").(string)))
+	addr := &computeBeta.Address{}
+	switch computeApiVersion {
+	case v1:
+		v1Address, err := config.clientCompute.Addresses.Get(
+			addressId.Project, addressId.Region, addressId.Name).Do()
+		if err != nil {
+			return handleNotFoundError(err, d, fmt.Sprintf("Address %q", d.Get("name").(string)))
+		}
+
+		err = Convert(v1Address, addr)
+		if err != nil {
+			return err
+		}
+	case v0beta:
+		v0BetaAddr, err := config.clientComputeBeta.Addresses.Get(
+			addressId.Project, addressId.Region, addressId.Name).Do()
+		if err != nil {
+			return handleNotFoundError(err, d, fmt.Sprintf("Address %q", d.Get("name").(string)))
+		}
+
+		err = Convert(v0BetaAddr, addr)
```

Comment:
@rosbo mentioned we don't actually need to convert here: https://github.com/terraform-providers/terraform-provider-google/pull/488/files#r142192681